app/src/main/java/com/example/pokemobil/ui/detail/DetailFragment.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.pokemobil.ui.detail
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.navArgs
+import com.example.pokemobil.databinding.FragmentDetailBinding
+import com.example.pokemobil.domain.model.Animated
+import com.example.pokemobil.domain.model.DetailPokemonModel
+import com.example.pokemobil.domain.extension.getUrl
+import com.example.pokemobil.domain.util.ServiceCountConst.MaxServiceCount
+import com.example.pokemobil.domain.util.ServiceCountConst.MinServiceCount
+import com.example.pokemobil.ui.extension.observe
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class DetailFragment : Fragment() {
+
+    private var _binding: FragmentDetailBinding? = null
+    private val binding get() = _binding!!
+    private val args: DetailFragmentArgs by navArgs()
+    private var pokemonId: Int = 0
+    private val viewModel: DetailViewModel by viewModels()
+    private var changePokemon = true
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        pokemonId = args.pokemonId
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentDetailBinding.inflate(inflater, container, false)
+        val view = binding.root
+        return view
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setObserve()
+        setOnClick()
+    }
+
+    private fun setObserve() {
+        viewModel.getPokemon(pokemonId)
+
+        observe(viewModel.success) {
+            setOnSuccess(it)
+        }
+        observe(viewModel.loading) {
```

Comment:
loading ve error için de birşeyler ekleyebiliriz ne dersin? Error için Toast mesajı olabilir. Loading için de progress indicator olabilir